messaging/src/main/java/org/axonframework/messaging/annotation/MessageHandlerInterceptorDefinition.java

```diff
@@ -89,6 +92,24 @@ public Object handle(Message<?> message, T target) throws Exception {
                     throw e;
                 });
             }
+            finally {
```

Comment:
Why not handle the case in the regular try block?

Reply:
...good point, I could've pointed that out to you, @YvonneCeelie. We can do this ASAP, after `chain.proceed`, since it's "just" attaching an operation to the `result` if it's a `CompletableFuture`. 
Doing that before we `return` is fine.